lib/LedgerSMB/Scripts/setup.pm

```diff
@@ -65,15 +66,15 @@ sub _get_database {
     my ($request) = @_;
     my $creds = $request->{_auth}->get_credentials('setup');
 
-    return [ 401,
+    return [ HTTP_UNAUTHORIZED,
              [ 'WWW-Authenticate' => 'Basic realm="LedgerSMB"',
                'Content-Type' => 'text/text; charset=UTF-8' ],
              [ 'Please enter your credentials' ] ]
         if ! defined $creds->{password};
 
     # Ideally this regex should be configurable per instance, and possibly per admin user
     # for now we simply use a fixed regex. It will cover many if not most use cases.
-    return [ 454,
+    return [ HTTP_454,
```

Comment:
@ziuq Hmm, 401 is unauthorized.
I think we may need to ask @ehuelsmann to ensure he is still happy with the choice of 454.
I actually added this code some time ago after discussion with ehuelsmann.
454 was chosen so we could present some specific information back to the client (Prohibited User)
It is not an standards assigned Status Code, however, it's a code that is unlikely to be assigned any time in the forseable future (there are many unused 4xx codes free below this one)

This is the  [Official HTTP Status Code Registry](https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml#http-status-codes-1)

Looking at that, and the more verbose, but incomplete wikipedia [List of HTTP Status Codes](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes) there really isn't a perfect match for our needs here.
However, if we were to change it, I'd suggest one of the "specifically cacheable" codes such as 404 which may assist with hardening against attacks.
This introduces issues we currently don't have a mechanism to resolve (in setup.pm) where the Client is unable to reliably display a useful error message to the User.

__NOTE: ANY CHANGE to this constant requires changes to be made in html and or .js files as well__

As for naming this constant, regardless of what number is used, I'd suggest
"HTTP_PROHIBITED_USERNAME" 
Even if we do later on change the actual response number this is descriptive and shouldn't need to be changed